Firestore/Example/Tests/SpecTests/FSTMockDatastore.mm

```diff
@@ -293,11 +295,12 @@ + (instancetype)mockDatastoreWithWorkerDispatchQueue:(FSTDispatchQueue *)workerD
   static DatabaseInfo database_info{DatabaseId{"project", "database"}, "persistence", "host",
                                     false};
 
-  static EmptyCredentialsProvider credentials;
+  static std::list<EmptyCredentialsProvider> credentials_providers;
+  credentials_providers.emplace_back();
```

Comment:
This now looks safe, but note that it "leaks" memory...  (True, the pointers are all still reachable... but nothing actually releases the memory until the program exits. It's just a test, so maybe that's not overly critical.)

In other words, this is roughly the same as just leaking the memory directly:

return [[FSTMockDatastore alloc] ... credentials:new EmptyCredentialsProvider()];

^ and that does have the benefit of making it more obvious.


If taking this approach, then we should at least add a comment; maybe something like:

// Note that we purposely don't bother to cleanup the EmptyCredentialsProvider instances

Reply:
done